### PR TITLE
fix: TwoHopSampler call

### PR DESCRIPTION
**Pull request by dekz**
## Description

`TwoHopSampler` needs to perform a `call` instead of a `staticcall`. The `amounts` need to be the last value returned by one of the sampler functions.

[Simbot](https://metabase.spaceship.0x.org/dashboard/186?run_id=jacob-fix-two-hop-sampler&adjusted_amount_bought_win_tolerance__bps_=0.5)

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts [hunk 1]
```diff
@@ -838,7 +841,10 @@ export class SamplerOperations {
                     };
                 });
             },
-            () => [],
+            () => {
+                logUtils.warn('SamplerContractOperation: Two hop sampler reverted');
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts:845]: Delete?

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts [hunk 2]
```diff
@@ -453,7 +453,10 @@ export const NATIVE_FEE_TOKEN_BY_CHAIN_ID = valueByChainId<string>(
     NULL_ADDRESS,
 );
 
-export const NATIVE_FEE_TOKEN_AMOUNT_BY_CHAIN_ID = valueByChainId({}, ONE_ETHER);
+export const NATIVE_FEE_TOKEN_AMOUNT_BY_CHAIN_ID = valueByChainId(
+    { [ChainId.Mainnet]: ONE_ETHER.times(0.1) },
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/src/utils/market_operation_utils/constants.ts:457]: ?

**Review by dorothy-zbornak — APPROVED** [review]

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts (hunk 2 above)
Comment (dekz) [reply packages/asset-swapper/src/utils/market_operation_utils/constants.ts:457]: Oh yeah, I felt like getting querying $4000 of the token value might be a bit high these days. WDYT

File: packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts (hunk 1 above)
Comment (dekz) [reply packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts:845]: I think keep so we can catch it when looking at logs, similar to the base sampler contract operation.

https://github.com/0xProject/protocol/blob/development/packages/asset-swapper/src/utils/market_operation_utils/sampler_contract_operation.ts#L59

File: packages/asset-swapper/src/utils/market_operation_utils/constants.ts (hunk 2 above)
Comment (dorothy-zbornak) [reply packages/asset-swapper/src/utils/market_operation_utils/constants.ts:457]: Yeah seems fair

**Pull request merged**